homeassistant/components/progettihwsw/config_flow.py

```diff
@@ -0,0 +1,102 @@
+"""Config flow for ProgettiHWSW Automation integration."""
+
+from ProgettiHWSW.ProgettiHWSWAPI import ProgettiHWSWAPI
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+
+from .const import DOMAIN  # pylint: disable=unused-import
+
+DATA_SCHEMA = vol.Schema(
+    {vol.Required("host"): str, vol.Required("port", default=80): int}
+)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user host input."""
+
+    api_instance = ProgettiHWSWAPI(f'{data["host"]}:{data["port"]}')
+    is_valid = await api_instance.check_board()
+
+    if is_valid is False:
+        raise CannotConnect
+
+    return {
+        "title": is_valid["title"],
+        "relay_count": is_valid["relay_count"],
+        "input_count": is_valid["input_count"],
+        "is_old": is_valid["is_old"],
+    }
+
+
+async def validate_input_relay_modes(data):
```

Comment:
This function isn't needed. Just set the allowed values in the form data schema and use `vol.In` as validator.